### ISSUE TITLE
Geospatial Taxonomy

### ISSUE THREAD

**Issue by cchristelis** [kartoza/healthyrivers#65]
# Problem

We need to attach data to a location. This can be (in this project) a site, a river reach, river, catchment area etc. But it could have a different breakdown in other biodiversity projects.

# Proposed Solution

Create a location/site/geospatial model, which
- [ ] can be configured as what type of spatial data describes it. [point/polygon/line ...] and 
- [ ] where it falls in the hierarchy of location data. (river is inside catchment area, site is inside river).

**Comment by timlinux** [comment]
Outputs:

- [ ] Schema diagram
- [ ] Models
- [ ] Tests
- [ ] Admin UI

**Comment by dimasciput** [comment]
Hi @cchristelis , does these models have any relationship with the legacy db here https://github.com/kartoza/healthyrivers/pull/74 ?

**Comment by cchristelis** [comment]
They will be new models @dimasciput, but we will do a data migration at some point to get the location data from the sites into these. (our view on a data record is that it is some record taken at some location, which could be a site, river etc) Although in this case it will probably conform 100% of the time to a site.

**Comment by dimasciput** [comment]
@timlinux @cchristelis 

I tried to made a diagram for these models: 

<img width="583" alt="screen shot 2018-02-14 at 3 18 51 pm" src="https://user-images.githubusercontent.com/1979569/36194015-ac61514c-119a-11e8-92b8-7288399632d2.png">


I made two models, which Type is what type of location is (river reach, river, or catchment), and the other one is the spatial data, they can be polygon or point or line. These two models have many to many relationship. Am I interpreting this correctly?

**Comment by cchristelis** [comment]
@dimasciput, I don't think we need a field for polygon/point/line ... there is a more generic field type that we can use and on save validate that it is correct according to the type. 

We'll need a foreignkey for Type from Site/Location. 

What are your thoughts behind the file_data field?
